### ISSUE TITLE
Design System Updates: Search/Filter Modal

### ISSUE THREAD

**Issue by annaseulgi** [hackforla/311-data#1600]
### Dependency 
- [X] https://github.com/hackforla/311-data/issues/1666
- [ ] #1848 
- [ ] https://github.com/hackforla/311-data/issues/1865

### Un-Blocks
- Dev Ticket: https://github.com/hackforla/311-data/issues/1868

### Overview
The design team needs to update the calendar picker to address the following issues identified by the research and product teams:
- No confirmation after users select a date range
- Layout of the calendar is not intuitive
- Data is limited to the past 30 days (NEW issue as of 2/21/24)

### Action Items
- [x] brainstorm w/ design team to identify which features to improve on the calendar picker
- [x] create wireframes and get feedback from design team
- [x] revise wireframes based on feedback and create interactive prototype
- [x] usability test interactive prototype with design team
- [x] run through interactive prototype with developers to identify potential constraints
- [x] add popover message to show users that data is limited to the past 30 days
- [ ] ~add double chevron to date picker/calendar to indicate earliest date of data collection~
- [ ] ~Remove "Last 3 Months" from the date range dropdown and replace with "Custom Date Range"~
- [x] handoff to developers

### Hand Off Materials

Figma Section Name: [Search/Filter Modal #1600 - Final](https://www.figma.com/design/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?node-id=11301-3834&node-type=section&t=nNeI8Sv2DemjiFeB-0)

<details><summary>Before Screenshot</summary>
<p>

<img width="343" alt="Screenshot 2024-10-02 at 6 20 28 PM" src="https://github.com/user-attachments/assets/b41e2eb8-c448-4100-b935-9ba49e5d1c84">
<img width="324" alt="Screenshot 2024-10-02 at 6 20 47 PM" src="https://github.com/user-attachments/assets/43318344-dce5-4747-9052-396ed1c13667">
<img width="324" alt="Screenshot 2024-10-02 at 6 21 04 PM" src="https://github.com/user-attachments/assets/01cbd49f-4ac6-4f83-ab11-c1c420281d3f">

</p>
</details> 

<details><summary>After Screenshot (Finalized)</summary>
<p>

<img width="259" alt="Screenshot 2024-10-02 at 6 22 15 PM" src="https://github.com/user-attachments/assets/baaeae52-6f72-4950-bb57-0aac3d33214e">
<img width="259" alt="Screenshot 2024-10-02 at 6 22 26 PM" src="https://github.com/user-attachments/assets/ac80cdc4-b20a-41a5-b4a9-a64fca466f33">
<img width="200" alt="Screenshot 2024-10-02 at 6 22 39 PM" src="https://github.com/user-attachments/assets/fb3de6c3-a4b2-4fbd-a23d-637dd66ff90c">

</p>
</details> 

### Resources/Instructions
- Interactive Prototype [Figma file](https://www.figma.com/file/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?type=design&node-id=11301-3834&mode=design&t=kQYC8cDa7D3k7ADY-0)
  - please refer to the 'Allison J' page and the wireframes under 'INTERACTIVE PROTOTYPE - 2.27.24'
- Popover Message [Figma file](https://www.figma.com/file/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?type=design&node-id=11219-3360&mode=design&t=kQYC8cDa7D3k7ADY-0)
  - refer to the 'Anna K' page and the 'Popover' section

**Comment by ryanfchase** [comment]
@allisonjeon @giraffe2003 

- we need to add a restriction to the calendar to limit to 30 days, need to add some messaging on how this is a limitation of a low-cost infrastructure

**Comment by cottonchristopher** [comment]
Ticket Outline/Review 

Title: 
Overview: we need X for Y 
- (try to keep to 1 sentence. What we want to do and why) 
- Y = the definition of done (objective criteria) 

Description/Details: (optional if needed)
- may include more of a description of the overview, 
- may include an image of the problem

Action Items: 
- tasks that can be checked off/completed. 
- [ ] make checkable

Resource/links: 
- include links and resources so that you don't have to search outside of the ticket for information
    - examples: Figma, links to other issues, resources, standards, and criteria used in completing the document. 
- Screenshots:  should be easily viewable, pertinent info (only) 
- Screenshots: before and after shots
    - make screenshots collapsable

**Comment by bberhane** [comment]
We'll need to update the last three action steps based on the dependency, as Dev is working to load multiple years of data

**Comment by cottonchristopher** [comment]
We are no longer limited to 30 days of data. However, we need to add an option to the date picker; Below "3 Month Range" add "Custom Date Range". In addition, we need to add the double chevron to indicate the beginning of the data

**Comment by bberhane** [comment]
Based on our PM meeting on 5-30-24 and for the purposes of the demo, we will be limiting data to 30 days, to avoid slow loading times. This ticket has been updated with new instructions and is ready for UX/UI design lead.

**Comment by annaseulgi** [comment]
assigning @allisonjeon and @giraffe2003  to this ticket for the modal redesign + updated date range picker

assigning myself for the popover component

**Comment by annaseulgi** [comment]
@allisonjeon @giraffe2003 hi all, for this ticket, I'm going to add an action item where part of the deliverable is to design the modal to look similar with the existing modal on the demo (other than what's being specifically changed to improve the modal of course). 

This is because the dev team is having confusion/trouble with distinguishing what's being handed off/changed when parts of the design do not look identical to what exists in the demo. Because our design system is a mess at the moment, sometimes the modal component being used in hand-offs doesn't look the same to what is already existing in the 311 site demo. 

The image below is what I screenshotted from the live site. If the modal for this ticket is being handed off and the only changes you want the dev team to make is the date range picker, then the rest of the design should look exact. If other parts are being intentionally changed, then no worries and that can be noted when everything is being handed off. Lmk if you have questions!

(tagging @ryanfchase for visibility)

![Image](https://github.com/hackforla/311-data/assets/140572118/b74083a4-8512-4da2-bf75-dba4c8eec204)

**Comment by ryanfchase** [comment]
>  If the modal for this ticket is being handed off and the only changes you want the dev team to make is the date range picker, then the rest of the design should look exact. If other parts are being intentionally changed, then no worries and that can be noted when everything is being handed off. 

@annaseulgi this is very helpful for the dev team, thank you for adding this as an action item. Agreed, if other parts change, we should note this on the ticket or in a comment

**Comment by ryanfchase** [comment]
Hi Angela,

Please leave a comment with the following items:

- updated ETA
- progress from the last week (if applicable)
- availability for communications during the week

**Comment by Joy-Truex** [comment]
Hi Angela,

Please leave a comment with the following items:

- updated ETA
progress from last week
- Any blockers this week?
- availability for communications during the week

Thanks!

**Comment by Joy-Truex** [comment]
Unassigned Angela who is no longer active with 311Data

**Comment by allisonjeon** [comment]
Hi @ryanfchase @bberhane for the last two checklist items (see below) - are these required before handing off the design to developers? Or is it something that can be implemented in another round of revisions to the search/filter modal?

- [ ]  add double chevron to date picker/calendar to indicate earliest date of data collection
- [ ]  Remove "Last 3 Months" from the date range dropdown and replace with "Custom Date Range"

The search/filter modal has been updated significantly to address these two original issues:
- No confirmation after users select a date range
- Layout of the calendar is not intuitive

If possible, I was wondering if we could add the two checklist items to ticket #1818 and move forward with the following design:
Interactive Prototype [Figma file](https://www.figma.com/file/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?type=design&node-id=11301-3834&mode=design&t=kQYC8cDa7D3k7ADY-0) - please refer to the 'Allison J' page and the wireframes under 'INTERACTIVE PROTOTYPE - 2.27.24'

Please let me know, thank you!

**Comment by ryanfchase** [comment]
@allisonjeon yep, works for me. I've moved those two action items to #1818, and I will strike them off this ticket. Good thinking, thanks!

**Comment by ryanfchase** [comment]
@allisonjeon I've also modified this ticket's original post with a `### Hand Off Materials` section, so that it is similar to our existing design tickets (we are no longer doing `Design to Engineering Hand Off` tickets).

Please provide any relevant "Before Screenshots" and "After Screenshots", either in the original comment (in their designated areas), or you can paste them in a new comment and I can help you format the ticket for hand off.

**Comment by allisonjeon** [comment]
Hey @ryanfchase I've updated the original comment with the relevant "Before Screenshots" and "After Screenshots". Please let me know what else is needed for hand off. Thank you!

**Comment by ryanfchase** [comment]
This is looking good @allisonjeon -- I'll have the PM team review it tomorrow evening. I'll adjust the labels and status to reflect "In Review", thanks 😄

**Comment by bberhane** [comment]
PMs have OK'ed this. G2G for dev

**Comment by traycn** [comment]
Engineering issue made for the Search/Filter Modal design update here: https://github.com/hackforla/311-data/issues/1868

**Comment by Joy-Truex** [comment]
- type to edit is a relatively light design lift
- dev solution will potentially allow for larger than 30 day range
- if that is the case then type to edit will be implemented

**Comment by traycn** [comment]
Iceboxed issue, dependant on https://github.com/hackforla/311-data/issues/1865

Dependency is on knowing how much data we are able to request. If 1865 resolves the app crashing when making a request for more than 30 days worth of data, we will be able to use a larger date range for the chips (i.e.: 7 Days, 14 Days, 30 Days).

**Comment by ryanfchase** [comment]
Question for design lead:
- [ ] I understand that the Blank map discussion may consider removing the chips from our search and filters menu. If design team decides to move forward with removing chips, can we remove chips on this design as well? This ticket is blocked due to feasibility of chips -- removing the chips would greenlight this ticket to hand off to dev

**Comment by Joy-Truex** [comment]
Will be discussed in tonight's breakout room after general meeting. ty!

**Comment by ryanfchase** [comment]
This ticket remains in the Questions column. I noticed not many designers were in attendance. If this needs to get kicked back to next general meeting, we can do that.

**Comment by ryanfchase** [comment]
@Joy-Truex should we move this ticket back to design's New Issue Approval so that the mockup should be adjusted to remove the chips? Or should this be considered blocked until an upcoming design ticket removes the chips? I'm guessing chips will be removed on the blank map ticket, so we can mark this ticket blocked by the blank map ticket:
- #1848

**Comment by Joy-Truex** [comment]
@ryanfchase Yes the chips will be removed as part of #1848 so in my opinion putting it back to new issue approval will be creating a duplicate task!

**Comment by ryanfchase** [comment]
Thanks @Joy-Truex, that's making perfect sense. I will move this to Dev, and I will just make a note NOT to include the date chips. I will instruct devs that the Datepicker component's padding will remain unchanged, aka we should match padding styles present in the "Before" screenshot (e.g. in #1600's References section). Since there is nothing left for design to do, we can consider this ticket closed. Feel free to re-open if I missed something!
- (WIP) Dev Ticket: https://github.com/hackforla/311-data/issues/1936
